### PR TITLE
build: remove usage of deprecated `::set-output` command for GHA

### LINKED ISSUE #10713: GHA Replace deprecated set-output & set-state calls
**Description**
Recently a lot of warning annotations occurred on our CI workflow due to the usage of the deprecated commands set-output & set-state, [see](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/).

### PR DESCRIPTION
**Pull request by lenaschoenburg**
This command is deprecated: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

closes #10713

### CODE REVIEW

File: .github/actions/build-zeebe/action.yml [hunk 1]
```diff
@@ -44,4 +44,4 @@ runs:
         mvn -B -DskipTests -DskipChecks install ${{ inputs.maven-extra-args }}
         export BUILD_DIR=$(mvn -pl dist/ help:evaluate -Dexpression=project.build.directory -q -DforceStdout)
         export ARTIFACT=$(mvn -pl dist/ help:evaluate -Dexpression=project.build.finalName -q -DforceStdout)
-        echo "::set-output name=result::${BUILD_DIR}/${ARTIFACT}.tar.gz"
+        echo "distball=result::${BUILD_DIR}/${ARTIFACT}.tar.gz" >> $GITHUB_OUTPUT
```
Comment (megglos) [inline .github/actions/build-zeebe/action.yml:47]: should be like this?
```suggestion
        echo "result=${BUILD_DIR}/${ARTIFACT}.tar.gz" >> $GITHUB_OUTPUT
```

Comment (megglos) [reply .github/actions/build-zeebe/action.yml]: renaming it to `distball` would also require adjusting line 26

Comment (oleschoenburg) [reply .github/actions/build-zeebe/action.yml]: Yep, already made the change :+1:

**Review by megglos — APPROVED** [review]
LGTM!

**Pull request merged**